src/main/scala/io/iohk/ethereum/consensus/validators/std/StdOmmersValidator.scala

```diff
@@ -66,16 +66,24 @@ class StdOmmersValidator(blockchainConfig: BlockchainConfig, blockHeaderValidato
   /** Validates that each ommer's header is valid based on validations stated in section 11.1 of the YP
     *
     * @param ommers               the list of ommers to validate
-    * @param getBlockHeaderByHash function to obtain ommers' parents
-    *
-    * @return [[OmmersValidator.OmmersValid]] if valid, an [[OmmersValidator.OmmersError.OmmersNotValidError]] otherwise
+    * @param getBlockParentsHeaderByHash function to obtain ommers' parents
+    * @return [[OmmersValidator.OmmersValid]] if valid, an [[OmmersValidator.OmmersError.OmmersHeaderError]] otherwise
     */
   private def validateOmmersHeaders(
       ommers: Seq[BlockHeader],
-      getBlockHeaderByHash: GetBlockHeaderByHash
+      getBlockParentsHeaderByHash: GetBlockHeaderByHash
   ): Either[OmmersError, OmmersValid] = {
-    if (ommers.forall(blockHeaderValidator.validate(_, getBlockHeaderByHash).isRight)) Right(OmmersValid)
-    else Left(OmmersNotValidError)
+    val validationsResult: Seq[Either[BlockHeaderError, BlockHeaderValid]] =
+      ommers.map(blockHeaderValidator.validate(_, getBlockParentsHeaderByHash))
+
+    if (validationsResult.forall(_.isRight)) Right(OmmersValid)
+    else {
+      val ommersHeaderError = validationsResult
+        .collect { case Left(error) => error }
+        .map(error => OmmersHeaderError(List(error)))
```

Comment:
`.map` and `.reduce` are not needed...can be simplified to
```
val errors = validationsResult.collect { case Left(error) => error }.toList
Left(OmmersHeaderError(errors))
```

Reply:
You are right, I really overcomplicated that :)

Reply:
:)